sklearn/model_selection/_split.py

```diff
@@ -1103,10 +1103,12 @@ def _find_best_fold(self, y_counts_per_fold, y_cnt, group_y_counts):
 class TimeSeriesSplit(_BaseKFold):
     """Time Series cross-validator.
 
-    Provides train/test indices to split time series data samples
-    that are observed at fixed time intervals, in train/test sets.
-    In each split, test indices must be higher than before, and thus shuffling
-    in cross validator is inappropriate.
+    Provides train/test indices to split time ordered data. Samples are
+    required to be equally spaced, for the folds to represent the same duration,
+    in order to have comparable metrics across folds.
+    With this split, shuffling in cross validator is inappropriate, otherwise some
+    train indices would be higher than test indices, representing learning from future
+    and evaluating on the past.
```

Comment:
Let's have a paragraph on generalities of time-ordered data and another one on good practices. Let's also make it clearer that not only shuffling is inappropriate, and that "the same duration" applies to test sets and not training sets.
```suggestion
    This method provides train/test indices for splitting time-ordered data,
    where other cross-validation methods are inappropriate as they would lead to
    training on future data and evaluating on past data.
    To ensure comparable metrics across folds, samples must be equally spaced.
    Once this condition is met, each test set covers the same time duration,
    while the train set size accumulates data from previous splits.
```